src/SBInclinedExponential.cpp

```diff
@@ -0,0 +1,286 @@
+        // count for the number of required digits, which is nominally given by digits10.
+        // cf. http://stackoverflow.com/questions/4738768/printing-double-without-losing-precision
+        // Unfortunately, there doesn't seem to be an easy equivalent of python's %r for
+        // printing the repr of a double that always works and doesn't print more digits than
+        // it needs.  This is the reason why we reimplement the __repr__ methods in python
+        // for all the SB classes except SBProfile.  Only the last one can't be done properly
+        // in python, so it will use the C++ virtual function to get the right thing for
+        // any subclass.  But possibly with ugly extra digits.
+        oss.precision(std::numeric_limits<double>::digits10 + 4);
+        oss << "galsim._galsim.SBInclinedExponential("<<getInclination()<<", "<<getScaleRadius()<<", "<<getScaleHeight();
+        oss <<", "<<getFlux()<<", False";
+        oss << ", galsim.GSParams("<<*gsparams<<"))";
+        return oss.str();
+    }
+
+    SBInclinedExponential::SBInclinedExponentialImpl::SBInclinedExponentialImpl(Angle inclination, double scale_radius,
+            double scale_height, double flux, const GSParamsPtr& gsparams) :
+        SBProfileImpl(gsparams),
+        _inclination(inclination),
+        _r0(scale_radius),
+        _h0(scale_height),
+        _flux(flux),
+        _inv_r0(1./_r0)
+    {
+        dbg<<"Start SBInclinedExponential constructor:\n";
+        dbg<<"inclination = "<<_inclination<<std::endl;
+        dbg<<"scale radius = "<<_r0<<std::endl;
+        dbg<<"scale height = "<<_h0<<std::endl;
+        dbg<<"flux = "<<_flux<<std::endl;
+
+        // Check if cos(inclination) is within allowed limits, and institute special handling if it isn't
+
+        double cosi = std::abs(_inclination.cos());
+
+        if(cosi<sbp::minimum_cosi)
+        {
+            // Perfectly edge-on isn't analytic, so we truncate at the minimum cos(inclination) value
+            cosi = sbp::minimum_cosi;
+        }
+
+        // Now set up, using this value of cosi
+
+        _r0_cosi = _r0*cosi;
+        _inv_r0_cosi = 1./_r0_cosi;
+
+        _h_tani_over_r = scale_height*std::abs(inclination.sin())*_inv_r0_cosi; // A tiny bit more accurate than using tan of
+                                                                                // the truncated value
+
+        _half_pi_h_tani_over_r = 0.5*M_PI*_h_tani_over_r;
+
+        xdbg<<"_h_tani_over_r = "<<_h_tani_over_r<<std::endl;
+        xdbg<<"_r0_cosi = "<<_r0_cosi<<std::endl;
+
+        // For large k, we clip the result of kValue to 0.
+        // We do this when the correct answer is less than kvalue_accuracy.
+        // (1+k^2 r0^2 cos^2(i))^-1.5 = kvalue_accuracy
+        // The convolution factor is generally a small effect, so we don't include it here
+        _ksq_max = (std::pow(this->gsparams->kvalue_accuracy,-1./1.5) - 1.)/cosi;
```

Comment:
I don't think this calculation is right.  First, a straightforward reading of the derivation would imply that you should have cosi**2 here.

But more importantly, this calculation too conservative.  Particularly in the case where cosi ~= 0.  In particular, it's not really appropriate to ignore the convolution factor, since it is significant precisely for ky >> 1, which is the limit you are taking for this calculation.  

I think you can probably do a few iterations of a non-linear solver to find where

(1+ky^2 cosi^2)**-1.5 * (_half_pi_h_sini_over_r * ky) / sinh(_half_pi_h_sini_over_r * ky) ~= epsilon

As I mentioned earlier, you'll want to do this separately for epsilon = kvalue_accuracy and maxk_threshold for two quantities _ksq_max and _maxk.  I think you'll find that you don't need to go out to nearly as high a ky value as this calculation would imply.  Particularly when cosi << 1.